src/_common/modal/AppModal.vue

```diff
@@ -54,10 +58,12 @@ onMounted(() => {
 		});
 	}
 
-	_beforeEachDeregister = router.beforeEach((_to, _from, next) => {
-		_dismissRouteChange();
-		next();
-	});
+	if (!props.ignoreRouter) {
```

Comment:
Put this within the router.beforeEach so that the prop can reactively change the behavior

Reply:
Changed to use `afterEach` and dismiss only if there was no reported navigation failure. This seemed to work well during testing for both dirty post forms and dirty comment forms.